### PR TITLE
Update torchvision classification tutorial

### PR DESCRIPTION
**Pull request by yutong-xiang-97**
## What has changed and why?

Use two separate datasets for pretraining and fine-tuning in the torchvision tutorial. Also download and process them separately.

## How has it been tested?

Running the code locally

[Classification with Torchvision’s ResNet - LightlyTrain documentation.pdf](https://github.com/user-attachments/files/20363430/Classification.with.Torchvision.s.ResNet.-.LightlyTrain.documentation.pdf)

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [x] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

File: docs/source/tutorials/resnet/index.md [hunk 1]
```diff
@@ -12,48 +12,170 @@ LightlyTrain can be installed directly from PyPI:
 pip install lightly-train
 ```
 
-## Dataset Preparation
+## Pretrain ResNet with LightlyTrain
+
+### Download the Pretraining Dataset
+
+Here we use [a dataset for human detection on Kaggle](https://www.kaggle.com/datasets/killa92/human-detection-dataset) for pretraining. It contains 170 PNG images (of which 146 for training and 24 for validation) of various sizes containing humans.
```
Comment (guarin) [inline docs/source/tutorials/resnet/index.md]: Should we write here that a small dataset is used for demonstration purposes but in reality one would use a much larger dataset?

File: docs/source/tutorials/resnet/index.md [hunk 2]
```diff
@@ -12,48 +12,170 @@ LightlyTrain can be installed directly from PyPI:
...
+
+You can download the dataset directly from Kaggle using the following commands (suppose you want the dataset to locate in `datasets`):
+
+```bash
+curl -L -o datasets/human-detection-dataset-pretraining.zip \
+https://www.kaggle.com/api/v1/datasets/download/killa92/human-detection-dataset
```
Comment (guarin) [inline docs/source/tutorials/resnet/index.md]: This command doesn't work on linux because `curl` doesn't create the `datasets` directory if it doesn't exist. There are two options:
* Add the `--create-dirs` flag to let curl create the directory
* Do not save the dataset into an extra `datasets` directory

File: docs/source/tutorials/resnet/index.md [hunk 3]
```diff
@@ -12,48 +12,170 @@ LightlyTrain can be installed directly from PyPI:
...
+```
+
+extract the zip file and rename the directory to `human-detection-dataset-pretraining`.
+
+```bash
+unzip datasets/human-detection-dataset-pretraining.zip -d datasets/ && \
+mv datasets/PennFudanPed datasets/human-detection-dataset-pretraining/
```
Comment (guarin) [inline docs/source/tutorials/resnet/index.md]: Let's only move the train images. This makes it clear that only unlabeled images are needed. 
```suggestion
mv datasets/PennFudanPed/train_images datasets/human-detection-dataset-pretraining
```

File: docs/source/tutorials/resnet/index.md [hunk 4]
```diff
@@ -12,48 +12,170 @@ LightlyTrain can be installed directly from PyPI:
...
+```bash
+tree -L 1 datasets/human-detection-dataset-pretraining
+```
+
+```bash
+> datasets/human-detection-dataset-pretraining
+> ├── train_images
+> └── train_masks
+> └── valid_images
+> └── valid_masks
```
Comment (guarin) [inline docs/source/tutorials/resnet/index.md]: Here we should only show the train images.

File: docs/source/tutorials/resnet/index.md (hunk 1 above)
Comment (guarin) [inline docs/source/tutorials/resnet/index.md]: I would drop the information about train and validation images. It is not relevant for pretraining. It will only confuse users.

File: docs/source/tutorials/resnet/index.md [hunk 5]
```diff
@@ -12,48 +12,170 @@ LightlyTrain can be installed directly from PyPI:
...
+Here we use [a dataset for human detection on Kaggle](https://www.kaggle.com/datasets/killa92/human-detection-dataset) for pretraining. It contains 170 PNG images (of which 146 for training and 24 for validation) of various sizes containing humans.
+
+You can download the dataset directly from Kaggle using the following commands (suppose you want the dataset to locate in `datasets`):
+
+```bash
+curl -L -o datasets/human-detection-dataset-pretraining.zip \
+https://www.kaggle.com/api/v1/datasets/download/killa92/human-detection-dataset
+```
+
+extract the zip file and rename the directory to `human-detection-dataset-pretraining`.
```
Comment (guarin) [inline docs/source/tutorials/resnet/index.md]:
```suggestion
Extract the zip file and rename the directory to `human-detection-dataset-pretraining`.
```

File: docs/source/tutorials/resnet/index.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 # Classification with Torchvision's ResNet
 
-This tutorial demonstrates how to pretrain a ResNet model from [Torchvision](https://pytorch.org/vision/stable/) using LightlyTrain and then fine-tune it for classification using the PyTorch Lightning framework. We will perform both steps on the [Human Detection Dataset](https://www.kaggle.com/datasets/constantinwerner/human-detection-dataset/data) from Kaggle.
+This tutorial demonstrates how to pretrain a ResNet model from [Torchvision](https://pytorch.org/vision/stable/) using LightlyTrain and then fine-tune it for classification using [PyTorch Lightning](https://lightning.ai/docs/pytorch/stable/). We will perform both steps on two different human detection datasets from Kaggle. Such datasets can be used for training models to detect humans in images, a basic task in industries like security and autonomous driving.
```
Comment (guarin) [inline docs/source/tutorials/resnet/index.md]: Maybe explain here that one dataset is used for pretraining and the other for fine-tuning. I would also mention that the pretraining one is unlabeled.

File: docs/source/tutorials/resnet/index.md [hunk 7]
```diff
@@ -227,7 +313,7 @@ from torchvision.models import resnet18
 from torchvision.transforms.v2 import Compose, Normalize, Resize, ToDtype, ToImage
 
 # Suppose you have the dataset in the datasets/ directory
-dataset_path = Path("datasets") / "human detection dataset"
+dataset_path = Path("datasets") / "human-detection-dataset-fine-tuning"
```
Comment (guarin) [inline docs/source/tutorials/resnet/index.md:309]: When I ran fine-tuning I didn't get any feedback regarding training or validation accuracy:
Screenshot 2025-05-21 at 10 32 27

I think this should be included somewhere. (Ran it on google colab).

**Review by guarin — CHANGES_REQUESTED** [review]

File: docs/source/tutorials/resnet/index.md (hunk 1 above)
Comment (yutong-xiang-97) [reply docs/source/tutorials/resnet/index.md]: Good point. Will add it

File: docs/source/tutorials/resnet/index.md [hunk 8]
```diff
@@ -12,48 +12,166 @@ LightlyTrain can be installed directly from PyPI:
...
+
+import matplotlib.pyplot as plt
+import numpy as np
+from PIL import Image
+
+# Suppose you have the dataset in the datasets/ directory
+dataset_path = Path("datasets") / "human-detection-dataset-pretraining"
+
+# Define paths to the training image directories
+train_data_path = dataset_path / "train_images"
```
Comment (guarin) [inline docs/source/tutorials/resnet/index.md]: Had to change it to this to work:
```suggestion
# Define paths to the training image directories
train_data_path = dataset_path
```

I think we can remove the `train_data_path` variable completely.

**Review by guarin — APPROVED** [review]
LGTM! Feel free to resolve and merge.
